.docker/search/uwsgi.ini

```diff
@@ -0,0 +1,17 @@
+# adapted from
+# - https://github.com/libero/search/blob/master/docker/uwsgi.ini
+[uwsgi]
+# Python modules integration
+wsgi-file=search/wsgi.py
+callable=APP
```

Comment:
Is there anyway that the details here can be read from the image? (ie just point to a file, rather than code inside a file)

Reply:
How much configuration options are there in `uwsgi.ini` that would be missed in that particular case? Maybe we could have an additional configuration `.ini` file instead that gets passed (and these options stay in the search repo)

Reply:
An option is that we ship the `uwsgi.ini` file; an override would still then need to rewrite it from scratch or to only use command line options like `--processes` (long-winded).

Reply:
`callable` is difficult to search for, rumor has it that there is a default value `application` but would be a very implicit way to integrate.

Reply:
you can make the uwsgi file itself callable. So, Rather than this:
```
from search import create_app

APP = create_app()
```
You can do this:
```
from search import create_app

if __name__ == "__main__":
    app = create_app()
    app.run()
```

Reply:
What would the `uwsgi.ini` file become? Omitting the `callable` configuration?

Reply:
Ok. Let's go back to the beginning. What's wrong with the current configuration?

Reply:
Nothing's wrong. Trying to reduce the knowledge here of the code (eLife Profiles changed the name (case) of the variable once, at it was a pain to do; in Libero it would be a breaking change). Easier to just refer to a file. (HTTP would be even easier...)

Reply:
@thewilkybarkid I think this is a valid concern.

@giorgiosironi If we remove `callable` are you suggesting that the settings in `uwsgi.ini` will be ignored?

Reply:
I'll experiment to check what happens if I remove `callable` altogether.

Reply:
Omitting the configuration line, or trying to unset it, leads to this error:
```
search_wsgi_1             | unable to find "application" callable in file search/wsgi.py
search_wsgi_1             | unable to load app 0 (mountpoint='') (callable not found or import error)
```

Also, `app.run()` in this context is not useful because it starts the local development web server on port 5000. `APP` is a callable (we get it by now) that accepts the two arguments `environ` and `start_response`. A reference to this object is [what uwsgi wants](https://www.python.org/dev/peps/pep-0333/#specification-details).

Reply:
[Special notation](https://github.com/elifesciences/lax-formula/blob/master/salt/lax/config/srv-lax-uwsgi.ini#L12):
```
module=core.wsgi:application
```
Only moves the problem into a single configuration key.

Reply:
It seems examples like https://github.com/justasabc/nginx_uwsgi_django/blob/84de28e052f30566dbc0348b8571aceb54a87fa6/gis_uwsgi.ini#L51 are relying on the [default value](https://docs.djangoproject.com/en/2.1/howto/deployment/wsgi/#the-application-object) being `application`. We could rename to that to be more in line with the standard de facto, but I'd still specify it here to avoid confusion as there is no clear documentation on uwsgi's default values.

Reply:
Let's have a look at this tomorrow in the office.